integrations/snyk/snyk/client.py

```diff
@@ -0,0 +1,268 @@
+from typing import Any, Optional, AsyncGenerator
+import httpx
+from loguru import logger
+
+from port_ocean.context.event import event
+
+
+class SnykClient:
+    def __init__(
+        self,
+        token: str,
+        api_url: str,
+        app_host: str | None,
+        org_id: str,
+        webhook_secret: str | None,
+    ):
+        self.token = token
+        self.api_url = f"{api_url}/v1"
+        self.app_host = app_host
+        self.org_id = org_id
+        self.rest_api_url = f"{api_url}/rest"
+        self.webhook_secret = webhook_secret
+        self.http_client = httpx.AsyncClient(headers=self.api_auth_header, timeout=30)
+        self.snyk_api_version = "2023-08-21"
```

Comment:
do we rely on the api version or this can be configured by the user? (in the azure exporter we allow the user to specify the api-version per resource kind )

Reply:
if its something that isnt realy changing i think itll be good enough for now IMO